src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -181,27 +181,31 @@ def get_paths_to_platform_and_hwsku_dirs():
 
     return (platform_path, hwsku_path)
 
-
-def get_path_to_port_config_file():
+def get_path_to_port_config_file(asic=None):
     """
     Retrieves the path to the device's port configuration file
 
     Returns:
         A string containing the path the the device's port configuration file
     """
-    # Get platform and hwsku path
     (platform_path, hwsku_path) = get_paths_to_platform_and_hwsku_dirs()
 
-    # First check for the presence of the new 'platform.json' file
-    port_config_file_path = os.path.join(platform_path, PLATFORM_JSON_FILE)
-    if not os.path.isfile(port_config_file_path):
-        # platform.json doesn't exist. Try loading the legacy 'port_config.ini' file
-        port_config_file_path = os.path.join(hwsku_path, PORT_CONFIG_FILE)
-        if not os.path.isfile(port_config_file_path):
-            raise OSError("Failed to detect port config file: {}".format(port_config_file_path))
+    port_config_candidates = []
+
+    # Check for 'platform.json' file presence first
+    port_config_candidates.append(os.path.join(platform_path, PLATFORM_JSON))
 
-    return port_config_file_path
+    # Check for 'port_config.ini' file presence in a few locations
+    port_config_candidates.append(os.path.join(hwsku_path, PORT_CONFIG_INI))
+    if asic:
+        port_config_candidates.append(os.path.join(hwsku_path, asic, PORT_CONFIG_INI))
+    port_config_candidates.append(os.path.join(platform_path, PORT_CONFIG_INI))
```

Comment:
I believe we should be able to simplify this as follows, but will wait for input from @samaity and @SuvarnaMeenakshi:

```python
    if asic:
        port_config_candidates.append(os.path.join(hwsku_path, asic, PORT_CONFIG_INI))
    else:
        port_config_candidates.append(os.path.join(hwsku_path, PORT_CONFIG_INI))
```

Reply:
Yes, the above suggested simplification looks good. If asic is defined, then platform/hwsku/port_config.ini will not be present as per current design.
In the previous code, I do not see port_config_candidates.append(os.path.join(platform_path, PORT_CONFIG_INI)). being directly used, and is used only if hwsku is not defined. Any reason for removing that check?

Reply:
> Yes, the above suggested simplification looks good. If asic is defined, then platform/hwsku/port_config.ini will not be present as per current design.

Great. Thanks for the confirmation.

> In the previous code, I do not see port_config_candidates.append(os.path.join(platform_path, PORT_CONFIG_INI)). being directly used, and is used only if hwsku is not defined. Any reason for removing that check?

Because `port_config.ini` has never resided under the `<platform>` path (unless something changed recently), as it has always been hardware SKU-specific. That's one of the reasons I'm asking these questions. I don't believe that is a valid check, and I want to confirm that it can be removed. @samaity: It appears you added this directory as a candidate. What was your reasoning?

Reply:
@jleveque , As I have understood till now and also during `platform.json` file porting, the order was like below.
```
/usr/share/sonic/hwsku/
/usr/share/sonic/device/<platform>/<hwsku>/
/usr/share/sonic/platform/<hwsku>/
/usr/share/sonic/<hwsku>/
```
So, I followed the same,  but yeah, I guess, I mistakenly added the directory. I don't think it's needed for `port_config.ini`. however, `platform.json` will be under the platform directory. So we are safe with the changes.

Reply:
So to finalize/summarize for posterity, `platform.json` will always be found under:
- `/usr/share/sonic/device/<platform>/` in the host OS
- `/usr/share/sonic/platform/` inside a container

And `port_config.ini` will always be found under:
- `/usr/share/sonic/device/<platform>/<hwsku>/<ASIC index>/` in the host OS of a multi-ASIC switch
- `/usr/share/sonic/device/<platform>/<hwsku>/` in the host OS of a single-ASIC switch
- `/usr/share/sonic/hwsku/<ASIC index>/` (also `/usr/share/sonic/platform/<hwsku>/<ASIC index>/`) in a container on a multi-ASIC switch
- `/usr/share/sonic/hwsku/` (also `/usr/share/sonic/platform/<hwsku>/`) in a container on a single-ASIC switch